### PR TITLE
Feature/razorpay payment integration

### PR DESCRIPTION
**Pull request by yashmaurya321-crypto**

### CODE REVIEW

File: controllers/UserController.js [hunk 1]
```diff
@@ -21,8 +29,8 @@ const registerUser = async (req, res) => {
        if (userExists) {
          return res.status(400).json({ error: "Email already exists" });
        }
-       const salt = await bcryptjs.genSalt(10);
-       const hashedPassword = await bcryptjs.hash(password, salt);
+       const salt = await bcrypt.genSalt(10);
```
Comment (geneus-solutions) [inline controllers/UserController.js:32]: why this was changed from bcryptjs to bcrypt?

File: routes/AuthRouter.js [hunk 2]
```diff
@@ -5,20 +5,20 @@ import User from "../models/User.js";
...
 const transporter = nodemailer.createTransport({
-  // Configure the email provider settings
-  // For example, for Gmail:
   service: "Gmail",
+ secure : true,
+ port: 465,  
   auth: {
-    user: process.env.email,
-    pass: process.env.password,
+    user: "",
```
Comment (geneus-solutions) [inline routes/AuthRouter.js:20]: why user and pass is set to empty string here?

File: routes/FoodRouter.js [hunk 3]
```diff
@@ -1,148 +1,15 @@
...
 
-    try {
-        const userExists = await User.findById(user);
-        if (!userExists) {
-            return res.status(404).json({ message: "User not found" });
-        }
+router.get('/api/food/:id',getFoodById);
 
-        let foodData = await Food.findOne({ user });
+router.put('/api/food', Auth, updateFood);
```
Comment (geneus-solutions) [inline routes/FoodRouter.js:12]: update route names to 
"/api/updateFood" -- update operation
"/api/addFood" -- update operation
"/api/removeFood" -- remove operation
"/api/getFoodById/:id" -- get food by id operation

File: routes/ItemRouter.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
+import { createItem, getItems } from '../controllers/ItemController.js';
 
-router.post('/', createItem)
-router.get('/', getItems)
+const router = express.Router();
 
-module.exports = router
\ No newline at end of file
+router.post('/', createItem);
+router.get('/api/item', getItems);
```
Comment (geneus-solutions) [inline routes/ItemRouter.js:7]: update route names to 
"/createFoodItem" -- create operation
"/getFoodItems" -- get operation

File: routes/User.js [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
+import express from 'express';
+import { getUser, registerUser, loginUser } from '../controllers/UserController.js';
+import { Auth, refreshTokenHandler } from '../controllers/AuthController.js';
+
+const router = express.Router();
+
+router.post('/api/user/register', registerUser);
+router.post('/api/user/login', loginUser);
+router.get('/api/user', Auth, getUser);
+router.post('/api/user/refresh', refreshTokenHandler);
```
Comment (geneus-solutions) [inline routes/User.js:10]: update refresh route to below
'api/user/refreshToken"

File: routes/DetailRouter.js [hunk 6]
```diff
@@ -1,7 +1,12 @@
-const express = require('express')
-const router = express.Router()
-const {updateDetail} = require('../controllers/DetailController')
-const Auth = require('../controllers/AuthController')
-router.post('/update', updateDetail)
+import express from 'express';
```
Comment (geneus-solutions) [inline routes/DetailRouter.js:1]: update this Router component name from DetailRouter to FoodDetailRouter..js

File: routes/DetailRouter.js [hunk 7]
```diff
@@ -1,7 +1,12 @@
...
+import { updateDetail } from '../controllers/DetailController.js';
+import { configDotenv } from 'dotenv';
+configDotenv()
+const router = express.Router();
 
-module.exports = router
\ No newline at end of file
+router.post('/api/detail/update', updateDetail);
+
+router.get('/api/procressenv', async (req, res) => {
```
Comment (geneus-solutions) [inline routes/DetailRouter.js:9]: why this router /api/procressenv'?

File: models/Items.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-const mongoose = require('mongoose');
+import mongoose from "mongoose";
```
Comment (geneus-solutions) [inline models/Items.js:1]: what is Items meaning here? if its food item, then rename to FoodItems.js

File: models/Details.js [hunk 9]
```diff
@@ -1,4 +1,4 @@
-const mongoose = require('mongoose');
+import mongoose from "mongoose";
```
Comment (geneus-solutions) [inline models/Details.js:1]: what is Details meaning? if its food details, then rename to FoodDetails.js

**Review by geneus-solutions — CHANGES_REQUESTED** [review]
remove acces.log file and Nutri-app-api.txt

**Comment by yashmaurya321-crypto** [comment]
ok i will change it

*Thanks & Regards,*

*Yash.S.Maurya*

*Mob - 7972273012*

**Pull request merged**